### PR TITLE
Fully move zulip_bots to Python 3.

### PR DESCRIPTION
**Pull request by fredfishgames**
Fixing issue #187

### CODE REVIEW

File: tools/custom_check.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def custom_check_file(fn, identifier, rules, skip_rules=None, max_length=None):
         # This rule might give false positives in virtualenv setup files which should be excluded,
         # and comments which should be rewritten to avoid use of "python2", "python3", etc.
         {'pattern': 'python[23]',
-         'exclude': set(['tools/provision']),
+         'exclude': set(['tools/provision', 'zulip_bots/']),
```
Comment (roberthoenig) [inline tools/custom_check.py]: I think you need to remove the `/` in `zulip_bots/` for this to work.

**Comment by fredfishgames** [comment]
Hi,
I made the changes you suggested. It still failed the tests...
I've also added a test to make sure python versions are python3

**Comment by fredfishgames** [comment]
Ok, I got rid of that test.. It was creating a bunch of false positives in comments etc.

**Comment by rht** [comment]
@fredfishgames see https://github.com/rht/python-zulip-api/commit/d11ad48342544f0bc10d8e8bd0dd5e3c8b7bb47c for the pattern needed for excluding `zulip_bots/` in 'python[23]' custom check. I ran the static analysis test locally and did not observe any failing.

**Comment by fredfishgames** [comment]
@rht Thanks! Didn't think of that

**Comment by showell** [comment]
@roberthoenig Can you merge this when you're happy with it and plan to be online for a while?  (I want to make sure you're available to revert it quickly if people have unexpected problems.)

**Comment by rht** [comment]
(Here is a case in zulip/zulip for having both `include_only` and `exclude` https://github.com/zulip/zulip/blob/4fd3492e737b38c79e781a24f1c0415cd824e6c7/tools/linter_lib/custom_check.py#L621-L633 -- also note there is a trailing slash at the end.)

**Comment by fredfishgames** [comment]
I've added a rule to the linter which makes sure any shebangs are python3, however I have not pushed it

**Comment by fredfishgames** [comment]
I'll be back in an hour to push it

**Comment by showell** [comment]
This commit got kind of stale, but I just merged a fix, borrowing the lint code from @fredfishgames and then manually fixing the lint errors myself (there were lots of conflicts due to recent activity, and we wanted to just remove shebangs from the test modules, so it was easiest just to re-do it).  Closing.

Thanks @fredfishgames!

**Pull request closed without merging**